### ISSUE TITLE
Unbound - fatal error: failed to setup modules - 24.1.2_3

### ISSUE THREAD

**Issue by ronin3510** [opnsense/core#7274]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [X] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [X] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**

Hi guys,

This is a bit of an oddity I'm seeing with Unbound on 1/4 machines that are now on 24.1.2_3

When I saw the service down I went to the logs and found this: 

===================

Error	unbound	Unable to open pipe. This is likely because Unbound isn't running.	
Notice	unbound	Backgrounding unbound logging backend.	
Critical	unbound	[74981:0] fatal error: failed to setup modules	
Error	unbound	[74981:0] error: module init for module python failed	
Error	unbound	[74981:0] error: pythonmod: python error: NoneType: None	
Error	unbound	[74981:0] error: pythonmod: can't parse Python script dnsbl_module.py	
Notice	unbound	daemonize unbound dhcpd watcher.	
Notice	unbound	[74981:0] notice: init module 0: python

===================

Restarted just in case - no change.

Reinstalled unbound - and made some progress. Service still shows down in GUI but appears operational in the console yet not resolving.

The logs:

============================
Informational	unbound	[76281:0] info: start of service (unbound 1.19.1).	
Notice	unbound	[76281:0] notice: init module 1: iterator	
Notice	unbound	[76281:0] notice: init module 0: validator	
Informational	unbound	[60338:0] info: server stats for thread 0: requestlist max 0 avg 0 exceeded 0 jostled 0	
Informational	unbound	[60338:0] info: server stats for thread 0: 0 queries, 0 answers from cache, 0 recursions, 0 prefetch, 0 rejected by ip ratelimiting	
Informational	unbound	[60338:0] info: service stopped (unbound 1.19.1).	
Informational	unbound	[60338:0] info: start of service (unbound 1.19.1).	
Notice	unbound	[60338:0] notice: init module 1: iterator	
Notice	unbound	[60338:0] notice: init module 0: validator

=====================================

And SSH output

==================================

root@OPNsense:~ # pkg install -f unbound
Updating OPNsense repository catalogue...
OPNsense repository is up to date.
Updating mimugmail repository catalogue...
mimugmail repository is up to date.
All repositories are up to date.
The following 1 package(s) will be affected (of 0 checked):

Installed packages to be REINSTALLED:
        unbound-1.19.1 [OPNsense]

Number of packages to be reinstalled: 1

1 MiB to be downloaded.

Proceed with this action? [y/N]: y
[1/1] Fetching unbound-1.19.1.pkg: 100%    1 MiB 334.8kB/s    00:04
Checking integrity... done (0 conflicting)
[1/1] Reinstalling unbound-1.19.1...
===> Creating groups.
Using existing group 'unbound'.
===> Creating users
Using existing user 'unbound'.
[1/1] Extracting unbound-1.19.1: 100%
You may need to manually remove /usr/local/etc/unbound/unbound.conf if it is no longer needed.
root@OPNsense:~ # service unbound onestart
Obtaining a trust anchor...
Starting unbound.
root@OPNsense:~ # drill isc.org @127.0.0.1
Error: error sending query: Could not send or receive, because of network error
root@OPNsense:~ # service unbound onestatus
unbound is running as pid 51697.

====================================================

So on one and the service appears to be running but not resolving in cli and not running in GUI

Unsure why this happens on 1/4 FWs, all are configured similarly, Unbound:53 for the OS with a port forward for AGH on all vlans. No recent changes to the configuration. 

Software version used and hardware type if relevant, e.g.:

OPNsense 24.1.2_3 (amd64).

**Comment by fichtner** [comment]
Ran into this today:

```
# /usr/local/sbin/unbound -dc /var/unbound/unbound.conf
Traceback (most recent call last):
  File "dnsbl_module.py", line 46, in <module>
    import dns.name
  File "/usr/local/lib/python3.9/site-packages/dns/name.py", line 27, in <module>
    import dns._features
  File "/usr/local/lib/python3.9/site-packages/dns/_features.py", line 3, in <module>
    import importlib.metadata
  File "/usr/local/lib/python3.9/importlib/metadata.py", line 9, in <module>
    import zipfile
  File "/usr/local/lib/python3.9/zipfile.py", line 6, in <module>
    import binascii
ImportError: Shared object "libz.so.6" not found, required by "binascii.cpython-39.so"
```

**Comment by fichtner** [comment]
probably from the last `snapshots/atest` pubished... please make sure to mention that...

**Comment by ronin3510** [comment]
Sorry about that, I mentioned 24.1.2_3 as I thought it only existed in snapshots at the time.

So I applied the patch, php looks happy now and reports the service as running but in fact it's not. You can see below I'm running -dc before and after the patch and the libz.so.6 missing dependency is still outstanding. I can fix it with a soft link but I don't really know why it showed up on this machine only and what's the proper fix

=======================================

root@OPNsense:~ # /usr/local/sbin/unbound -dc /var/unbound/unbound.conf
Traceback (most recent call last):
  File "dnsbl_module.py", line 46, in <module>
    import dns.name
  File "/usr/local/lib/python3.9/site-packages/dns/name.py", line 27, in <module>
    import dns._features
  File "/usr/local/lib/python3.9/site-packages/dns/_features.py", line 3, in <module>
    import importlib.metadata
  File "/usr/local/lib/python3.9/importlib/metadata.py", line 9, in <module>
    import zipfile
  File "/usr/local/lib/python3.9/zipfile.py", line 6, in <module>
    import binascii
ImportError: Shared object "libz.so.6" not found, required by "binascii.cpython-39.so"
root@OPNsense:~ #
root@OPNsense:~ #
root@OPNsense:~ # opnsense-patch fd00972
Fetched fd00972 via https://github.com/opnsense/core
Hmm...  Looks like a unified diff to me...
The text leading up to this was:
--------------------------
|From fd00972988b96a9a4ff2582bfe72ba3b7fbcc6ee Mon Sep 17 00:00:00 2001
|From: Franco Fichtner <franco@opnsense.org>
|Date: Sat, 24 Feb 2024 18:34:48 +0100
|Subject: [PATCH] unbound: fix missing /lib; closes #7274
|
|---
| src/etc/inc/plugins.inc.d/unbound.inc | 1 +
| 1 file changed, 1 insertion(+)
|
|diff --git a/src/etc/inc/plugins.inc.d/unbound.inc b/src/etc/inc/plugins.inc.d/unbound.inc
|index f74ba58e78b..5c9f52d3bed 100644
|--- a/src/etc/inc/plugins.inc.d/unbound.inc
|+++ b/src/etc/inc/plugins.inc.d/unbound.inc
--------------------------
Patching file etc/inc/plugins.inc.d/unbound.inc using Plan A...
Hunk #1 succeeded at 142.
done
All patches have been applied successfully.  Have a nice day.
root@OPNsense:~ # /usr/local/sbin/unbound -dc /var/unbound/unbound.conf
Traceback (most recent call last):
  File "dnsbl_module.py", line 46, in <module>
    import dns.name
  File "/usr/local/lib/python3.9/site-packages/dns/name.py", line 27, in <module>
    import dns._features
  File "/usr/local/lib/python3.9/site-packages/dns/_features.py", line 3, in <module>
    import importlib.metadata
  File "/usr/local/lib/python3.9/importlib/metadata.py", line 9, in <module>
    import zipfile
  File "/usr/local/lib/python3.9/zipfile.py", line 6, in <module>
    import binascii
ImportError: Shared object "libz.so.6" not found, required by "binascii.cpython-39.so"
root@OPNsense:~ # service unbound onestatus
unbound is not running.
root@OPNsense:~ #

root@OPNsense:~ # ls -l /usr/lib/libarchive.
libarchive.a     libarchive.so.7  libarchive.so@
root@OPNsense:~ # ln -s /usr/lib/libarchive.so /usr/lib/libz.so.6

root@OPNsense:~ # /usr/local/sbin/unbound -dc /var/unbound/unbound.conf
[1708797988] unbound[6217:0] error: bind: address already in use
[1708797988] unbound[6217:0] fatal error: could not open ports

root@OPNsense:~ # service unbound onestart
Obtaining a trust anchor...
Starting unbound.
root@OPNsense:~ # service unbound onestatus
unbound is running as pid 48773.
root@OPNsense:~ #


=======================================

**Comment by fichtner** [comment]
It needs an apply, not a (re)start.
